components/scream/src/physics/shoc/shoc_iso_c.f90

```diff
@@ -0,0 +1,102 @@
+module shoc_iso_c
+  use iso_c_binding
+  implicit none
+
+#include "scream_config.f"
+#ifdef SCREAM_DOUBLE_PRECISION
+# define c_real c_double
+#else
+# define c_real c_float
+#endif
+
+!
+! This file contains bridges from scream c++ to shoc fortran.
+!
+
+contains
+  subroutine append_precision(string, prefix)
+
+    character(kind=c_char, len=128), intent(inout) :: string
+    character(*), intent(in) :: prefix
+    real(kind=c_real) :: s
+
+    write (string, '(a,i1,a1)') prefix, sizeof(s), C_NULL_CHAR
+  end subroutine append_precision
+
+  subroutine shoc_init_c(nlev, gravit, rair, rh2o, cpair, &
+                         zvir, latvap, latice, karman, &
+                         pref_mid, nbot_shoc, ntop_shoc) bind(c)
+    use shoc, only: shoc_init
+
+    integer(kind=c_int), value, intent(in) :: nlev ! number of levels
+
+    real(kind=c_real), value, intent(in)  :: gravit ! gravity
+    real(kind=c_real), value, intent(in)  :: rair   ! dry air gas constant
+    real(kind=c_real), value, intent(in)  :: rh2o   ! water vapor gas constant
+    real(kind=c_real), value, intent(in)  :: cpair  ! specific heat of dry air
+    real(kind=c_real), value, intent(in)  :: zvir   ! rh2o/rair - 1
+    real(kind=c_real), value, intent(in)  :: latvap ! latent heat of vaporization
+    real(kind=c_real), value, intent(in)  :: latice ! latent heat of fusion
+    real(kind=c_real), value, intent(in)  :: karman ! Von Karman's constant
+
+    real(kind=c_real), intent(in), dimension(nlev) :: pref_mid ! reference pressures at midpoints
+
+    integer(kind=c_int), value, intent(in) :: nbot_shoc ! Bottom level to which SHOC is applied
+    integer(kind=c_int), value, intent(in) :: ntop_shoc ! Top level to which SHOC is applied
+
+    call shoc_init(nlev, gravit, rair, rh2o, cpair, &
```

Comment:
Here I think you need the full
```F90
    pref_mid = 0
    call shoc_init(nlev, gravit, rair, rh2o, cpair, zvir, latvap, latice, karman, &
         pref_mid, nlev, 1)
    npbl = nlev ! set pbl layer explicitly so we don't need pref_mid.
```
from https://github.com/E3SM-Project/scream-docs/blob/master/shoc-port/shoc_c_intr.F90 so there aren't unit'ed access to pmid and so that npbl gets init'ed.